src/include/replication/replication_messages.h

```diff
@@ -30,16 +33,105 @@ namespace noisepage::replication {
 ENUM_DEFINE(ReplicationMessageType, uint8_t, REPLICATION_MESSAGE_TYPE_ENUM);
 #undef REPLICATION_MESSAGE_TYPE_ENUM
 
+class MessageFacade;
+DEFINE_JSON_HEADER_DECLARATIONS(MessageFacade);
+
+/** Abstraction over the underlying format used to send replication messages over the network */
+class MessageFacade {
+ public:
+  /** The underlying format of messages used in replication */
+  using MessageFormat = common::json;
+
+  /** Default constructor */
+  MessageFacade() = default;
+
+  /** Constructor using underlying message format */
+  explicit MessageFacade(MessageFormat underlying_message) : underlying_message_(std::move(underlying_message)) {}
```

Comment:
thoughts on `MessageFormat &&underlying_message`?

Reply:
essentially, makes the copy a bit more explicit for the caller. I don't think I have a strong opinion here since we're hopefully replacing the JSON.

Reply:
Turns out this constructor wasn't used so I removed it.